task_pool.go

```diff
@@ -0,0 +1,141 @@
+package getty
+
+import (
+	"fmt"
+	"sync"
+	"sync/atomic"
+)
+
+import (
+	log "github.com/AlexStocks/log4go"
+)
+
+//func init() {
+//	rand.Seed(time.Now().UnixNano())
+//}
+//
+//var (
+//	// The golang rand generators are *not* intrinsically thread-safe.
+//	randIDLock sync.Mutex
+//	randIDGen  = rand.New(rand.NewSource(time.Now().UnixNano()))
+//)
+//
+//func randID() uint64 {
+//	randIDLock.Lock()
+//	defer randIDLock.Unlock()
+//
+//	return uint64(randIDGen.Int63())
+//}
+
+// task t
+type task struct {
+	session *session
+	pkg     interface{}
+}
+
+// task pool: manage task ts
+type taskPool struct {
+	idx    int32
+	qLen   int32 // task t queue length
+	size   int32 // pool size
+	qArray []chan task
+	wg     sync.WaitGroup
+
+	once sync.Once
+	done chan struct{}
+}
+
+// build a task pool
+func newTaskPool(poolSize int32, taskQLen int32) *taskPool {
+	return &taskPool{
+		size:   poolSize,
+		qLen:   taskQLen,
+		qArray: make([]chan task, poolSize),
+		done:   make(chan struct{}),
+	}
+}
+
+// start task pool
+func (p *taskPool) start() {
+	if p.size == 0 {
+		panic(fmt.Sprintf("[getty][task_pool] illegal pool size %d", p.size))
+	}
+
+	if p.qLen == 0 {
+		panic(fmt.Sprintf("[getty][task_pool] illegal t queue length %d", p.qLen))
+	}
+
+	for i := int32(0); i < p.size; i++ {
+		p.qArray[i] = make(chan task, p.qLen)
+		p.wg.Add(1)
+		taskID := i
+		p.run(int(taskID))
```

Comment:
I think it should be ` go p.run(int(taskID))`

Reply:
Yeah, u r right. There should be a "go".